File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testdata/sync/CacheableSyncOneMany.java
package ru.tinkoff.kora.cache.annotation.processor.testdata.sync;

import ru.tinkoff.kora.cache.annotation.CacheInvalidate;
import ru.tinkoff.kora.cache.annotation.CachePut;
import ru.tinkoff.kora.cache.annotation.Cacheable;
import ru.tinkoff.kora.cache.annotation.processor.testcache.DummyCache11;
import ru.tinkoff.kora.cache.annotation.processor.testcache.DummyCache12;

import java.math.BigDecimal;

public class CacheableSyncOneMany {

    public String value = "1";

    @Cacheable(DummyCache11.class)
    @Cacheable(DummyCache12.class)
    public String getValue(String arg1) {
        return value;
    }

    @CachePut(value = DummyCache11.class, parameters = {"arg1"})
    @CachePut(value = DummyCache12.class, parameters = {"arg1"})
    public String putValue(BigDecimal arg2, String arg3, String arg1) {
        return value;
    }

    @CacheInvalidate(DummyCache11.class)
    @CacheInvalidate(DummyCache12.class)
    public void evictValue(String arg1) {

    }

    @CacheInvalidate(value = DummyCache11.class, invalidateAll = true)
    @CacheInvalidate(value = DummyCache12.class, invalidateAll = true)
    public void evictAll() {

    }
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testdata/CacheableSyncOneMany.kt
package ru.tinkoff.kora.cache.symbol.processor.testdata

import ru.tinkoff.kora.cache.annotation.CacheInvalidate
import ru.tinkoff.kora.cache.annotation.CachePut
import ru.tinkoff.kora.cache.annotation.Cacheable
import ru.tinkoff.kora.cache.symbol.processor.testcache.DummyCache1
import ru.tinkoff.kora.cache.symbol.processor.testcache.DummyCache12
import java.math.BigDecimal

open class CacheableSyncOneMany {
    var value = "1"

    @Cacheable(value = DummyCache1::class)
    @Cacheable(value = DummyCache12::class)
    open fun getValue(arg1: String): String {
        return value
    }

    @CachePut(value = DummyCache1::class, parameters = ["arg1"])
    @CachePut(value = DummyCache12::class, parameters = ["arg1"])
    open fun putValue(arg2: BigDecimal?, arg3: String?, arg1: String): String {
        return value
    }

    @CacheInvalidate(value = DummyCache1::class)
    @CacheInvalidate(value = DummyCache12::class)
    open fun evictValue(arg1: String) {
    }

    @CacheInvalidate(value = DummyCache1::class, invalidateAll = true)
    @CacheInvalidate(value = DummyCache12::class, invalidateAll = true)
    open fun evictAll() {
    }
}
